modules/netaddress.go

```diff
@@ -36,10 +36,10 @@ func (na NetAddress) IsLoopback() bool {
 	}
 
 	host := na.Host()
-	if ip := net.ParseIP(host); ip != nil && ip.IsLoopback() {
+	if ip := net.ParseIP(host); ip != nil && (ip.IsLoopback() || ip.IsUnspecified()) {
 		return true
 	}
-	if host == "localhost" || host == "::" || host == "0.0.0.0" {
+	if host == "localhost" {
```

Comment:
we had these here because of explicit quirks found in production. There should have been a comment explaining that, but you can't remove these like that, you probably want to replace them with a function.

We should also have a regression test for these changes.